pycloudlib/azure/cloud.py

```diff
@@ -357,6 +357,16 @@ def _create_vm_parameters(self, vm_name, image_id, nic_id, user_data):
         vm_parameters["storage_profile"][
             "image_reference"] = util.get_image_reference_params(image_id)
 
+        # We can have pro images from two differente sources, marktplaces
+        # and snapshots. A snapshot image do not have encoded in their
+        # image id the necessary info to create the 'plan' dict. To solve
+        # that, we are getting the necessary info from the registered_images
+        # dict, were we store the required information
+        registered_image = self.registered_images.get(image_id)
```

Comment:
The way we are setting up and using registered_images is really just to keep track of the original image_id on snapshots right? What do you think about instead adding a tag "img_src:<image_id>" for any snapshot created by pycloudlib? Would we then be able determine that information from snapshot metadata in the future via az SDK calls checking tags on an image (if the image_id isn't of  the form "publisher:offer:sku"?